### PR TITLE
Update design-patterns.mdx

### LINKED ISSUE #1713: Documentation provides incorrect code sample
I just started reading the documentation from scratch and noticed this problem: the section [Cadence Design Patterns / Script-Accessible report](https://docs.onflow.org/cadence/design-patterns/#script-accessible-report) provides a sample code with incorrect return. Instead of 

```cadence
// Return the struct with a script
pub fun main(): A.BReport {
    let b: AContract.BResource // Borrow the resource
    let report = b.generateReport()
    return b
}
```
The comment suggests the actual code should be

```cadence
// Return the struct with a script
pub fun main(): A.BReportStruct {
    let b: AContract.BResource // Borrow the resource
    let report = b.generateReport()
    return report
}
```
I would also directly return the response of `b.generateReport` directly instead of assigning it to a variable. 
I can raise a pull request to fix this myself, just need a green light here.

### PR DESCRIPTION
**Pull request by mfahimasim**
Closes #1713

## Description

Updated "Script-Accessible report" example code.
Updated grammar and punctuation mistakes.

### CODE REVIEW

File: docs/design-patterns.mdx [hunk 1]
```diff
@@ -139,23 +139,23 @@ pub contract AContract {
...
-import A from 0xA
+import AContract from 0xAContract
 
 // Return the struct with a script
-pub fun main(): A.BReport {
+pub fun main(): AContract.BReportStruct {
     let b: AContract.BResource // Borrow the resource
     let report = b.generateReport()
-    return b
+    return report
```
Comment (turbolent) [inline docs/design-patterns.mdx:158]: Could just be `return b.generateReport()`

**Review by turbolent — APPROVED** [review]
Thank you for improving the documentation!

**Comment by mfahimasim** [comment]
> Thank you for improving the documentation!

Welcome

File: docs/design-patterns.mdx (hunk 1 above)
Comment (mfahimasim) [reply docs/design-patterns.mdx:158]: absolutely

**Pull request merged**